File: core/common/src/main/java/ru/alexeyoss/core/common/di/scope/PerActivity.java
package ru.alexeyoss.core.common.di.scope;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Scope;

/**
 * Activity-level dagger scope
 * This scope placed between {@link PerApplication} and {@link PerScreen}
 */
@Scope
@Retention(RetentionPolicy.RUNTIME)
public @interface PerActivity {
}

File: core/common/src/main/java/ru/alexeyoss/foodie/core/common/di/scope/PerActivity.kt
package ru.alexeyoss.foodie.core.common.di.scope

import javax.inject.Scope

/**
 * Activity-level dagger scope
 * This scope placed between [PerApplication] and [PerScreen]
 */
@Scope
@Retention(AnnotationRetention.RUNTIME)
annotation class PerActivity 